### ISSUE TITLE
`raw` of `MLDOMAttribute` and `MLDOMPreprocessorSpecificAttribute` are different

### ISSUE THREAD

**Issue by JounQin** [markuplint/markuplint#168]
I'm developing a custom [markuplint-angular-parser](https://github.com/rx-ts/markuplint/tree/main/packages/angular-parser) which parses correctly IMO.

But in the following example, rule `attr-duplication` does not work at all.

```html
<input type="text" [type]="'password'" />
```

I tried to debug and found the `name.raw` of [`MLDOMPreprocessorSpecificAttribute`](https://github.com/markuplint/markuplint/blob/master/packages/@markuplint/ml-core/src/ml-dom/tokens/preprocessor-specific-attribute.ts#L25) is different from [`MLDOMAttribute`](https://github.com/markuplint/markuplint/blob/master/packages/@markuplint/ml-core/src/ml-dom/tokens/attribute.ts#L76).

I'm not familiar with the other preprocessors, I was thinking they should check `node.potential` instead of `node.raw` in `attr-duplication`.

And I think we should provide `name` and `value` property/`MLToken` in `MLASTPreprocessorSpecificAttr` like `MLASTHTMLAttr`.

**Comment by JounQin** [comment]
`id-duplication` won't work as my expectation too, `MLDOMPreprocessorSpecificAttribute` will just be ignored.

https://github.com/markuplint/markuplint/blob/master/packages/%40markuplint/rules/src/id-duplication/index.ts#L15

**Comment by YusukeHirao** [comment]
@JounQin Hi Joun.

It is may not info you expected but,
FYI: 
https://github.com/markuplint/markuplint/blob/master/packages/%40markuplint/vue-parser/src/attr.ts
https://github.com/markuplint/markuplint/blob/master/packages/%40markuplint/svelte-parser/src/attr.ts

**Comment by JounQin** [comment]
@YusukeHirao Then should I parse dynamic attributes as `MLASTHTMLAttr` instead of `MLASTPreprocessorSpecificAttr`? May I ask what does `MLASTPreprocessorSpecificAttr` mean?

(Actually current document is not enough for developers. 😂)

**Comment by YusukeHirao** [comment]
@JounQin 

`MLASTPreprocessorSpecificAttr` is used on pug-parser when a markup author wrote `.className` and/or `#idName`.
It is used only on `pug-parser` for now.

It should use `isDynamic` if an attribute in the square bracket on Angular.

**Comment by JounQin** [comment]
@YusukeHirao Thanks, then the next question is why the rules are checking `attr.raw` instead of `attr. potential`? In my example `<input type="text" [type]="'password'" />`, the second attribute `raw` should be `[type]` but `potential` should `type`, right?

**Comment by YusukeHirao** [comment]
@JounQin 

For instance, in Vue:
```vue
<input v-bind:type="password" />
```
`v-bind:type` is `type` potentially. It uses to evaluate the potential attribute name. In for example, `attr-duplication`, `deprecated-attr`, `id-duplication`, `invalid-attr`, `permitted-contents`, `required-attr`, and `wai-aria`.

Umm... Maybe, Did I wrong the word "potential"?

**Comment by JounQin** [comment]
@YusukeHirao 

Then should `<input type="text" v-bind:type="password" />` emit error on `attr-duplication`?

On current implementation, `attr.raw` is used, it will not emit error.

**Comment by YusukeHirao** [comment]
Oh, shoot!
It's [using `attr.raw`](https://github.com/markuplint/markuplint/blob/master/packages/%40markuplint/rules/src/attr-duplication/index.ts#L17).
It is my mistake😱
